src/transformers/generation_utils.py

```diff
@@ -2275,6 +2351,8 @@ def beam_search(
             )
             if model_kwargs["past"] is not None:
                 model_kwargs["past"] = self._reorder_cache(model_kwargs["past"], beam_idx)
+                if return_dict_in_generate:
+                    past_key_values += (model_kwargs["past"],)
```

Comment:
Assumes that if `model_kwargs["past"` is not `None` -> `output_past_key_values == True`